packaging/installer/methods/openwrt.md

```diff
@@ -0,0 +1,60 @@
+<!--
+title: "Install Netdata on OpenWRT"
+description: "Install Netdata on OpenWRT to monitor the health and performance of routers with thousands of real-time, per-second metrics."
+custom_edit_url: https://github.com/netdata/netdata/edit/master/packaging/installer/methods/openwrt.md
+sidebar_label: "OpenWRT"
+learn_status: "Published"
+learn_rel_path: "Installation/Install on specific environments"
+-->
+
+# Install Netdata on OpenWRT
+
+> 💡 This document may not be completely up-to-date. Please double-check the
+> details of the installation process before proceeding, particularly in 
+> places where specific software or package versions are mentioned.
+>
+> You can help improve this document by [submitting a
+> PR](https://github.com/netdata/netdata/edit/master/packaging/installer/methods/openwrt.md) with your recommended
+> improvements or changes. Thank you!
+
+Since PRs [#14822](https://github.com/netdata/netdata/pull/14822) and [#14994](https://github.com/netdata/netdata/pull/14994), it is possible to install Netdata static builds on OpenWRT using [kickstart.sh](https://github.com/netdata/netdata/blob/master/packaging/installer/methods/kickstart.md). However, some extra steps are required to ensure a successful Netdata install, with features such as log rotation and automatic updates working correctly.
+
+A new installation of OpenWRT [may not offer enough space](https://forum.openwrt.org/t/howto-resizing-root-partition-on-x86/140631#the-root-cause-1) for the Netdata static builds to be decompressed and for the installation to be completed successfully. In that case, the root partition and the root filesystem may need to be expanded. A comprehensive guide on how to accomplish that on x86 can be found [here](https://forum.openwrt.org/t/howto-resizing-root-partition-on-x86/140631#the-root-cause-1).
+
+As of Netdata version `v1.39.0`, at least 300MB of space are required for the root filesystem, if `dbengine` is disabled. If `dbengine` is used to store metrics locally, a larger root partition is required.
```

Comment:
I am afraid that this is too much! How many routers supported by OpenWrt have such requirements?

- https://openwrt.org/supported_devices

The least requirements for devices, which are supported, are 8MB storage, and 64 RAM.

Reply:
Us requiring some large amount of space than the minimum supported by OpenWrt is perfectly reasonable. Pretty much every piece of software out there that lists some kind of disk space requirement lists _more_ than the platform it’s running on quotes as a minimum requirement.

And, for the record, there’s no way that we’re ever going to be able to support the smallest systems that OpenWrt runs on. If you need data collection there, SNMP is the way to go, not a native Netdata install.

Reply:
I think you got me wrong on this. Nobody is saying that you should support the smallest systems. Of course, applications are growing, but 300 MB is just huge. You are counting that you will be supporting __only x86 systems__ on OpenWrt. That's a small number of routers that are using it. You can check the statistics: https://downloads.openwrt.org/stats/ (``Downloads (Top 10)``, but also check ``OpenWrt firmware image downloads``). The most downloaded image is for Xiaomi Mi Router 4A, which I do have to test, but the storage is there 64 MB, not much to install Netdata. Not many routers at least a few years ago did not have microSD slot so they can boot from it. 

I just want to pinpoint that currently, in each OpenWrt version, the Netdata installable package has 7,8 MB. Yes, that's true that it is for version 1.33.1, but your requirements are growing unexpectedly higher.

Reply:
> You are counting that you will be supporting only x86 systems on OpenWrt.

This isn't true, we are trying to provide best support for all architectures that we are currently [offering static builds for](https://learn.netdata.cloud/docs/installation/installation-methods/static-binary-packages) (64-bit x86, ARMv7, AArch64, and POWER8+ at the moment). Unfortunately and for various reasons, it is not possible to strip down our static builds and exclude certain plugins to reduce space requirements, as it is done in the [OpenWrt Netdata package Makefile](https://github.com/openwrt/packages/blob/master/admin/netdata/Makefile). 

However, we appreciate we could be doing a better job in supporting resource-constrained devices, especially when it comes to disk and RAM usage. We had an internal discussion on this topic today and I will create a feature request for us to work towards a stripped-down Netdata install for such environments.

In the meantime, in https://downloads.openwrt.org/stats/awstats.downloads.openwrt.org.downloads.html you can see that RPis are in the top 10 hits - that's just one example where this guide applies to. Additionally, this PR was created due to https://github.com/netdata/netdata/issues/9517, so there are valid cases to justify the purpose of it.

Reply:
I appreciate that Netdata is open source hwoever it doesn't help when libraries are being bundled without any simple way of unbundling. This also goes against most packaging policies for various distros in general. On a positive not Netdata don't depend on that many libraries or very uncommon ones so overall footprint very large however realistically you have to give up very dinky devices. I'm not sure what the internal policy is when it comes to packaging but I'm quite sure most distros would cooperative if you were to submit improvements to downstream which would benefit everyone in the end.

Reply:
I saw your comments, All the parties involved have good points. Probably the best way is to publish this document with two major sections. One is to install Netdata via the package manager (community supported packages) and one via the Netdata static builds (kickstart install). In each section, we will acknowledge the pros and cons of each approach so the openWRT community could get the best out of it. 

What I would like to mention is, @BKPepe @diizzyy how could I help you better support the community efforts?  I mean you ship a pretty old Netdata version and I think there are improvements that could benefit the openWRT users. Feel free to ping me and collaborate on that.

Reply:
@tkatsoulas 
Apart from the web frontend issue stop hardcoding bundled variants of libraries such as sqlite, h2o, dlib or at least provide an option to use external variants.

Reply:
Not sure if it's still of interest but since the move to CMake it's a bit easier to unbundle libraries / sources from external projects.

Having a quick look dlib is easy, h2o(lib) shouldn't be that hard. h2o upstream however bundles a bunch of libraries themselves too. There seems possibly some non upstreamable changes done to (lib)judy or at least breaking changes compared to the forked repo ( https://github.com/netdata/libjudy )? Sqlite seems a bit more tightly integrated in netdata but I didn't look closely at it.